src/Dimension.h

```diff
@@ -83,7 +83,7 @@ class Dimension {
     /** Get a different dimension of the same buffer */
     // @{
     Dimension dim(int i);
-    const Dimension dim(int i) const;
+    Dimension dim(int i) const;
```

Comment:
Wait... why do both these accessors exist if the const one is just as good as the non-const one?

Reply:
They probably don't. I'll just delete the non-const one. The claim is: "readability-const-return-type:
Checks for functions with a const-qualified return type and recommends removal of the const keyword. Such use of const is usually superfluous, and can prevent valuable compiler optimizations."